rholang/src/main/scala/coop/rchain/rholang/interpreter/Reduce.scala

```diff
@@ -241,8 +272,10 @@ object Reduce {
                                         substituteChannel[M].substitute(pattern)(1, env))
                     } yield (BindPattern(substPatterns, rb.remainder, rb.freeCount), q))
         // TODO: Allow for the environment to be stored with the body in the Tuplespace
-        substBody <- substitutePar[M].substitute(receive.body.get)(0, env.shift(receive.bindCount))
-        _         <- consume(binds, substBody, receive.persistent)
+        substBody <- substitutePar[M].substituteNoSort(receive.body.get)(
```

Comment:
Because nothing is gained by sorting the body. We don't need it in canonical form, we just need it in a deterministic form, and so substituting without re-sorting is still deterministic.